core/src/main/scala/kafka/admin/ReassignPartitionsCommand.scala

```diff
@@ -1609,14 +1610,15 @@ object ReassignPartitionsCommand extends Logging {
   }
 
   def parsePartitionReassignmentData(jsonData: String): (Seq[(TopicPartition, Seq[Int])], Map[TopicPartitionReplica, String]) = {
-    Json.parseFull(jsonData) match {
-      case Some(js) =>
+    Json.tryParseFull(jsonData) match {
+      case Right(js) =>
         val version = js.asJsonObject.get("version") match {
           case Some(jsonValue) => jsonValue.to[Int]
           case None => EarliestVersion
         }
         parsePartitionReassignmentData(version, js)
-      case None => throw new AdminOperationException("The input string is not a valid JSON")
+      case Left(f) =>
+        throw f
```

Comment:
Will the calling code handle this exception? Or do we have to wrap it under `AdminOperationException`?

Reply:
In this case I deliberately don't handle the exception in order to propagate it to the user who called this command. The purpose of this is to propagate the JSON parsing error so they know why (and where) was the JSON invalid. I can wrap this in an AdminOperationException but since it is intended for the user and not for catching, I think it can be left as it is.

Reply:
I think it would be more consistent to wrap it in `AdminOperationException`. We throw them in `parsePartitionReassignmentData` too. I agree it doesn't make a huge difference, but it's slightly better than diverging here for no clear reason.

Reply:
Ok